pkg/storage-init/storage-init.sh

```diff
@@ -202,9 +203,12 @@ if P3=$(findfs PARTLABEL=P3) && [ -n "$P3" ]; then
     if [ "$P3_FS_TYPE" = zfs ]; then
         set_sequential_mdev
         if ! chroot /hostfs zpool import -f persist; then
-            echo "$(date -Ins -u) Cannot import persist pool on P3 partition $P3 of type $P3_FS_TYPE, recreating it as $P3_FS_TYPE_DEFAULT"
-            INIT_FS=1
-            P3_FS_TYPE="$P3_FS_TYPE_DEFAULT"
+            # Don't re-format the fs if we are in kdump kernel
+            if [ "$IS_IN_KDUMP_KERNEL" = 0 ]; then
```

Comment:
Could we instead determine that zfs is not supported by the kernel e.g., by looking explicitly for that, or is there a unique exit code from zpool import in this case?

Reply:
Only by circumstantial evidence. there is no module. or an error is returned. in any case, I would not do a file system reformat from the kdump kernel